### DISCUSSION TITLE
Question on how to learn GNOME Extensions

### DISCUSSION THREAD

**Discussion by ghost** [Schneegans/Fly-Pie#308] [Fly-Pie Q&A]
Hi Simon! May I ask where you would recommend me first learning how to code GNOME extensions? Ultimately, I want to learn about how the Fly-Pie Integration works on GNOME so that I can implement a new feature for the clipboard where it can work with the GNOME screenshot tool and autosort clipboard contents, or even locate HTML content on webpages based on bounding boxes (similar to the firefox clipping tool). Thanks in advance for your time!

**Comment by Schneegans** [comment]
Hi there! Sorry for not answering earlier, I somehow missed this :sweat_smile: 

For getting started, I would suggest to look at all the resources linked [on this page](https://wiki.gnome.org/Projects/GnomeShell/Extensions). Especially [the GJS guide](https://gjs.guide/extensions/development/creating.html) is pretty good. Whenever you are stuck, you can ask your questions in the [extension matrix channel](https://matrix.to/#/#extensions:gnome.org). There are usually very helpful folks around!
